string deviceid
int8 state